src/test/lib/optimizer/optimizer_test.cpp

```diff
@@ -331,4 +337,43 @@ TEST_F(OptimizerTest, OptimizesSubqueriesExactlyOnce) {
   }
 }
 
+TEST_F(OptimizerTest, CheckTrueCacheablility) {
+  auto optimizer = Optimizer::create_default_optimizer();
+
+  auto lqp = ProjectionNode::make(expression_vector(add_(b, subquery_a)),
```

Comment:
Can those be `const`?

Reply:
No, they can't. 
If I try to make optimizer const, the build fails with the following message: `error: passing 'const hyrise::Optimizer' as 'this' argument discards qualifiers`.
If I try to make lqp const, the optimizer wouldn't have the only ownership over the query plan which it needs and the test fails.